README.md

```diff
@@ -21,147 +24,131 @@ or to use in an npm script
-2017-07-12T18:56:35.610Z TRACE Push successful
-2017-07-12T18:56:39.139Z INFO build nodejs-rest-http-s2i-1 complete
-2017-07-12T18:56:39.147Z INFO docker image repository 172.30.1.1:5000/nodeshift-demo/nodejs-rest-http
-2017-07-12T18:56:39.171Z INFO creating deployment configuration nodejs-rest-http
-2017-07-12T18:56:39.177Z INFO creating new serivce nodejs-rest-http
-2017-07-12T18:56:39.178Z INFO creating new route nodejs-rest-http
-2017-07-12T18:56:39.187Z INFO route host mapping nodejs-rest-http-nodeshift-demo.192.168.99.100.nip.io
-2017-07-12T18:56:39.199Z INFO done
-```
+Along with the command line, there is also a public API.  The API mirrors the commands.
 
-If you login to your local Openshift cluster and navigate to the "node-demo" project, there should be a new deployment with a route to your application.
+* resource
 
-For example: `http://nodejs-rest-http-node-demo.192.168.99.100.nip.io/`
+* applyResource
 
-_please note: Currently, once a route, service, deployment config, build config, and imagestream config are created, those are re-used. The only thing that changes from deployment to deployment is the source code.  This will change in the future_
+* build
 
-### .Nodeshift Directory
+* deploy
 
-The `.nodeshift` directory is responsible for holding your resource files.  These are `.yml` files that describe you service, deployments and routes.
+* undeploy
 
-Currently, nodeshift will only create resources based on the files specified,  in the future, its possible somethings could be created by default
+Options that you can specify on the command line, can also be passed as an options object to the API
 
-#### Commands & Goals
+#### Example Usage
 
-By default, if you run just `nodeshift`, it will run the `deploy` goal, which runs everything.
+    const nodeshift = require('nodeshift');
 
-**resource** - will parse and create the application resources files on disk
+    // Deploy an Application
+    nodeshift.deploy().then((response) => {
+        console.log(response);
+        console.log('Application Deployed')
+    }).catch((err) => {
+        console.log(err);
+    })
 
-**apply-resource** - does the resource goal and will also push that to your running cluster
+_please note: Currently, once a route, service, deployment config, build config, and imagestream config are created, those are re-used. The only thing that changes from deployment to deployment is the source code.  For application resources, you can update them by undeploying and then deploying again.  BuildConfigs and Imagestreams can be re-created using the --build.recreate flag_
 
-**deploy** - does everything, the archiving of code and everything that apply-resource does
+## Advanced Options
 
-**undeploy** - removes things that were deployed with the resource command
+While nodeshift tries to be very opinionated during the deployment process, but there are options available to pass to the cli or the API.
```

Comment:
This sentence is awkward. How about

While nodeshift is very opinionated about deployment parameters, both the CLI and the API accept options that allow you to customize nodeshift's behavior.

Reply:
english is hard :)